cmd/dlt/operatorrole/cmd.go

```diff
@@ -154,43 +118,109 @@ func run(cmd *cobra.Command, argv []string) {
 			os.Exit(1)
 		}
 	}
+
+	clusterKey := ""
+	var foundOperatorRoles []string
 	var spin *spinner.Spinner
 	if r.Reporter.IsTerminal() {
 		spin = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 	}
-	if spin != nil {
-		r.Reporter.Infof("Fetching operator roles for the cluster: %s", clusterKey)
-		spin.Start()
-	}
+	fetchingReporterOutput := "Fetching operator roles for the"
+	if args.prefix == "" {
+		clusterKey = r.GetClusterKey()
+		r.Reporter.Debugf("Loading cluster '%s'", clusterKey)
+		sub, err := r.OCMClient.GetClusterUsingSubscription(clusterKey, r.Creator)
+		if err != nil {
+			if errors.GetType(err) == errors.Conflict {
+				r.Reporter.Errorf("More than one cluster found with the same name '%s'. Please "+
+					"use cluster ID instead", clusterKey)
+				os.Exit(1)
+			}
+			r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
+			os.Exit(1)
+		}
+		if sub != nil {
+			clusterKey = sub.ClusterID()
+		}
+		c, err := r.OCMClient.GetCluster(clusterKey, r.Creator)
+		if err != nil {
+			if errors.GetType(err) != errors.NotFound {
+				r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
+				os.Exit(1)
+			} else if sub == nil {
+				r.Reporter.Errorf("Failed to get cluster '%s': %v", r.ClusterKey, err)
+				os.Exit(1)
+			}
+		}
 
-	isHypershift := false
-	if c != nil {
-		isHypershift = c.Hypershift().Enabled()
+		if c != nil && c.ID() != "" {
+			r.Reporter.Errorf("Cluster '%s' is in '%s' state. Operator roles can be deleted only for the "+
+				"uninstalled clusters", c.ID(), c.State())
+			os.Exit(1)
+		}
+		isHypershift := false
+		if c != nil {
+			isHypershift = c.Hypershift().Enabled()
```

Comment:
Why do we enable `Hypershift` if the cluster is not `nil`?

Reply:
It retrieves the value that the cluster is using, not necessarily enabled it